### PR TITLE
CHE-1785: fix problems with the tree of changes on Preview window

### PR DESCRIPTION
**Pull request by svor**
### What does this PR do?

Improves the widget which shows structure of the refactoring's changes
### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/1786
### Tests written?

No

Please review @vparfonov.

### CODE REVIEW

File: plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/refactoring/preview/PreviewViewImpl.java [hunk 1]
```diff
@@ -166,11 +180,100 @@ public void onSelectionChange(SelectionChangeEvent event) {
...
+            createTreeElement(treeItem, child.getText(), child.getChildrens());
+            root.addItem(treeItem);
+        }
+    }
+
+    private void checkParentState(TreeItem root, Boolean value) {
+        if (root.getParentItem() == null) {
+            return;
+        }
+        if (value && !((CheckBox)(((FlowPanel)root.getParentItem().getWidget()).getWidget(0))).getValue()) {
```
Comment (tolusha) [inline plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/refactoring/preview/PreviewViewImpl.java]: Could you simplify pls if it is possible

Comment (svor) [inline plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/refactoring/preview/PreviewViewImpl.java]: Yes, no problem. I have cleaned.

**Comment by vparfonov** [comment]
ok

File: plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/refactoring/preview/PreviewViewImpl.java (hunk 1 above)
Comment (tolusha) [inline plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/refactoring/preview/PreviewViewImpl.java]: thx.

**Pull request merged**